arangod/Aql/AqlValue.cpp

```diff
@@ -928,40 +960,80 @@ int AqlValue::Compare(velocypack::Options const* options, AqlValue const& left,
 
   switch (leftType) {
     case VPACK_INLINE_INT64:
-    case VPACK_INLINE_UINT64:
-      // if right value type is also optimized inline we can optimize comparison
-      if (leftType == rightType) {
-        if (leftType == VPACK_INLINE_UINT64) {
-          uint64_t l = static_cast<uint64_t>(left.toInt64());
-          uint64_t r = static_cast<uint64_t>(right.toInt64());
-          if (l == r) {
-            return 0;
+      switch (rightType) {
+        case VPACK_INLINE_INT64:
+          return comp<int64_t>(left.asInt64(), right.asInt64());
+        case VPACK_INLINE_UINT64:
+          return VelocyPackHelper::compareInt64UInt64(left.asInt64(),
+                                                      right.asUInt64());
+        case VPACK_INLINE_DOUBLE: {
+          double d = right.asDouble();
+          if (std::isnan(d)) [[unlikely]] {
+            return basics::VelocyPackHelper::cmp_less;
```

Comment:
What if both `left` and `right` are NaNs? Then I think their values must compare equal.